apps/src/code-studio/pd/application/teacher1920/Teacher1920Application.jsx

```diff
@@ -21,6 +21,15 @@ export default class Teacher1920Application extends FormController {
 
   static sessionStorageKey = 'Teacher1920Application';
 
+  componentDidMount() {
+    window.addEventListener('beforeunload', event => {
+      if (!this.state.submitting) {
```

Comment:
Should we do anything different when state is submitting?

Reply:
No; I'm not sure what you have in mind.

We intentionally do nothing in response to this event when `submitting == true`, so the event falls through to the default behavior: _Not_ to interrupt navigation to the thank-you page after successful submission.

Reply:
Just curious if somebody might interrupt the submission while the spinner is showing, and whether we might want to protect against that.

Reply:
Good point - will do this in a follow-up PR.